### ISSUE TITLE
Cannot read/write /etc/hosts with no-ns-hosts option

### ISSUE THREAD

**Issue by GRomR1** [dlenski/vpn-slice#127]
I want prevent any writes into /etc/hosts and I use `--no-host-names --no-short-names --no-ns-hosts` options. But I always get a message `WARNING: Couldn't configure hosts provider: Cannot read/write /etc/hosts`. I think when this options was added this configuration doesn't need. 

What I want? I want to use [vpn-slice](https://github.com/dlenski/vpn-slice) without any hidden sudo privileges on my MacOS. Only add `%admin  ALL=(ALL) NOPASSWD: /opt/homebrew/bin/vpn-slice` to sudoers file.

**Comment by dlenski** [comment]
> I want prevent any writes into /etc/hosts and I use `--no-host-names --no-short-names --no-ns-hosts` options.

`--no-short-names` has no additional effect beyond `--no-host-names`.

> What I want? I want to use [vpn-slice](https://github.com/dlenski/vpn-slice) without any hidden sudo privileges on my MacOS.

What are "hidden" sudo privileges?

`vpn-slice` has to be able to configure _routing_. If it can't add at least one route over the tunnel interface, then it's **completely useless**: you might as well just use `openconnect --script '/bin/true'`.

And on *BSD and Mac, configuring routing means [invoking `/sbin/route`](https://github.com/dlenski/vpn-slice/blob/master/vpn_slice/mac.py#L32-L35) as `root`.

So I don't understand what you're trying to do here.

**Comment by GRomR1** [comment]
@dlenski sorry if I was not be clear enough. 

But you has point me a right direction. Thank you!

My problem was not in vpn-slice. I've tried to run openconnect  without `sudo` and with `/bin/true` instead of `vpn-slice` and I got this error:
```
...
[2023-07-15 23:12:15] Session authentication will expire at Sun Jul 16 23:12:14 2023
[2023-07-15 23:12:15] Failed to connect utun unit: Operation not permitted
[2023-07-15 23:12:15] Failed to open tun device: Operation not permitted
[2023-07-15 23:12:15] Set up tun device failed
[2023-07-15 23:12:15] Send BYE packet: Set up tun device failed
[2023-07-15 23:12:15] Unrecoverable I/O error; exiting.
```

After that I found [this](https://www.infradead.org/openconnect/nonroot.html) guide in the openconnect site. It describes  the point if I want to run openconnect via ordinary user I should create a `tun` device previously. But there is a big problem to make a new `tun` device because I can't found any solution except [this](https://tuntaposx.sourceforge.net/) (`tuntaposx`).  The project (`tuntaposx`) was abandoned.


I created this issue because I thought the my problem was vpn-slice that can't be run as non-root user. But the problem has in macos network subsystem, no way to create a tun device without root priveligies.

Can I close the issue?
